custom_libs/subliminal_patch/providers/jimaku.py

```diff
@@ -0,0 +1,421 @@
+        else:
+            episode_number = video.episode
+
+        return {
+            'anidbid': candidate_anime.get('anidbid'),
+            'episode_number': episode_number
+        }
+    
+    @cache
+    def _assemble_jimaku_search_url(self, video, series_name):
+        """
+        Return a search URL for the Jimaku API.
+        Will first try to determine an Anilist ID based on properties of the video object.
+        If that fails, will simply assemble a query URL that relies on Jimakus fuzzy search instead.
+        """
+        
+        # Attempt to derive anilist_id
+        derived_anilist_id = None
+        derived_anidb_id = None
+        tag_to_derive_from = ""
+        
+        logger.info(f"Attempting to derive anilist ID...")
+        for tag in ["series_anidb_episode_id", "series_anidb_id", "tvdb_id", "series_tvdb_id"]:
+            candidate = getattr(video, tag, None)
+            
+            if candidate:
+                # Because tvdb assigns a single ID to all seasons of a show, we'll have to use another list to determine the correct AniDB ID
+                if "tvdb" in tag and video.season > 1:
+                    tvdb_anidb_mapping = self._get_tvdb_anidb_mapping(video, candidate)
+                    derived_anidb_id = tvdb_anidb_mapping['anidbid']
+                    if tvdb_anidb_mapping['episode_number'] != video.episode:
+                        self.video_episode_number_override = tvdb_anidb_mapping['episode_number']
+                    
+                    logger.info(f"Found AniDB ID '{derived_anidb_id}' for TVDB ID '{candidate}', season '{video.season}'")
+                    break
+                    
+                tag_to_derive_from = tag
+                logger.info(f"Got candidate tag '{tag_to_derive_from}' with value '{candidate}'")
+                break
+
+        if tag_to_derive_from or derived_anidb_id:
+            try:
+                url = "https://raw.githubusercontent.com/Fribb/anime-lists/master/anime-list-mini.json"
+                anime_list = self._webrequest_with_cache(url)
+                
+                if derived_anidb_id:
+                    mapped_tag = "anidb_id"
+                    value_to_use = derived_anidb_id
+                else:
+                    # Left: video, right: anime-lists
+                    tag_map = {
+                        "series_anidb_id": "anidb_id",
+                        "series_anidb_episode_id": "anidb_id",
+                        "tvdb_id": "thetvdb_id",
+                        "series_tvdb_id": "thetvdb_id",
+                    }
+                    mapped_tag = tag_map[tag_to_derive_from]
+                    value_to_use = getattr(video, tag_to_derive_from)
+                
+                obj = [obj for obj in anime_list if mapped_tag in obj and str(obj[mapped_tag]) == str(value_to_use)]
+                logger.debug(f"Based on '{mapped_tag}': '{value_to_use}', anime-list matched: {obj}")
```

Comment:
I feel like adding the anilist_id should be a refiner similar as the anidb so it can be reused by other providers too

Reply:
Added refiner in [1af8b4f](https://github.com/morpheus65535/bazarr/pull/2505/commits/1af8b4fda534cfe16c88c00870db434fa8811dfe)